ext/cosmos/ext/structure/structure.c

```diff
@@ -901,6 +901,11 @@ static VALUE binary_accessor_write(VALUE self, VALUE value, VALUE param_bit_offs
 
       if (given_bit_size <= 0) {
         end_bytes = -(given_bit_size / 8);
+        /* If passed a huge negative bit_size we need to truncate end_bytes
+           to the size of the passed buffer to prevent overflow. */
+        if (end_bytes > buffer_length) {
+          end_bytes = buffer_length;
+        }
```

Comment:
This should have been caught at check_bounds_and_buffer_size().   If the buffer is not big enough for the end bytes then it is too small overall and an error.